### DISCUSSION TITLE
flow "Auto ADB over TCPIP + Shizuku" stopped by failure

### DISCUSSION THREAD

**Discussion by brit-60** [rahaaatul/automate-flows#10] [General]
Hi. I'm new with automate and trying to start wifi debug and shizuku at Android boot (flow "Auto ADB over TCPIP + Shizuku"), but came across this following issue: 

System setting set
12-21 00:30:27.291 F 9@4: android.os.RemoteException: Remote stack trace:
	at com.android.providers.settings.SettingsProvider.enforceHasAtLeastOnePermission(SettingsProvider.java:3101)
	at com.android.providers.settings.SettingsProvider.mutateGlobalSetting(SettingsProvider.java:2150)
	at com.android.providers.settings.SettingsProvider.insertGlobalSetting(SettingsProvider.java:2104)
	at com.android.providers.settings.SettingsProvider.call(SettingsProvider.java:715)
	at android.content.ContentProvider.call(ContentProvider.java:2760)

12-21 00:30:27.291 I 9@4: Stopped by failure

I have followed the instructions here: https://github.com/rahaaatul/automate-flows/wiki/Auto-ADB-Mode

Anyone to help me solve this?

**Comment by rahaaatul** [comment]
**This is not a section to post issues.**

Check if you have any unchecked permission and if you have wireless debugging available in your device.

**Reply by brit-60** [reply]
Hello rahaaatul

I've tried running the flow once more and got a different mistake

12-21 14:49:42.389 I 19@1: Flow beginning
12-21 14:49:42.391 I 19@10: Wi-Fi network connected?
12-21 14:49:42.401 I 19@4: System setting set
12-21 14:49:42.404 I 19@5: Network service discover
12-21 14:49:45.417 I 19@15: Expression true?
12-21 14:49:45.419 I 19@12: Failure catch
12-21 14:49:45.421 I 19@6: ADB protocol set
12-21 14:49:45.482 F 19@6: android.system.ErrnoException: isConnected failed: ECONNREFUSED (Connection refused)
12-21 14:49:45.492 I 19@16: System setting set
12-21 14:49:45.497 I 19@17: Delay
12-21 14:49:46.509 I 19@4: System setting set
12-21 14:49:46.519 I 19@5: Network service discover
12-21 14:49:49.538 I 19@15: Expression true?
12-21 14:49:49.541 I 19@12: Failure catch
12-21 14:49:49.544 I 19@6: ADB protocol set
12-21 14:49:49.599 F 19@6: android.system.ErrnoException: isConnected failed: ECONNREFUSED (Connection refused)
12-21 14:49:49.610 I 19@16: System setting set
12-21 14:49:49.614 I 19@17: Delay
12-21 14:49:50.625 I 19@4: System setting set
12-21 14:49:50.633 I 19@5: Network service discover
12-21 14:49:52.625 I 19@5: Stopped by user

I'm running automate with Android 14 and such version provides dynamic ports everytime I restart wifi debug.

Considering the https://github.com/rahaaatul/automate-flows/wiki/Auto-ADB-Mode instructions I wonder whether they are for Android 14?

I'm sorry to have started an issue in the wrong session. Please, which session should I have used? 

Thanks for answering.

**Reply by rahaaatul** [reply]
That's not android 14 specific.

You need to first, go to automate settings and change the mode adb. See if it works. If that works, the flow will too.

> I'm sorry to have started an issue in the wrong session. Please, which session should I have used?

[Issues](https://github.com/rahaaatul/automate-flows/issues) page.

**Comment by brit-60** [comment]
Thanks

Em sáb., 21 de dez. de 2024, 16:07, Rahatul Ghazi ***@***.***>
escreveu:
